src/infra/core/src/utils/datasets_filtering.rs

```diff
@@ -171,11 +171,11 @@ pub fn matches_remote_ref_pattern(
 
 ////////////////////////////////////////////////////////////////////////////////
 
-pub fn get_local_datasets_stream(
-    dataset_repo: &dyn DatasetRepository,
+pub fn get_local_datasets_stream<'a>(
+    dataset_repo: &'a dyn DatasetRepository,
     dataset_ref_patterns: Vec<DatasetRefAnyPattern>,
-    current_account_name: AccountName,
-) -> impl Stream<Item = Result<DatasetRefAny, GetDatasetError>> + '_ {
+    current_account_name: &AccountName,
+) -> impl Stream<Item = Result<DatasetRefAny, GetDatasetError>> + 'a {
```

Comment:
Maybe passing `Arc` would be less wordy?

Reply:
`impl Stream` saves us an allocation and additional level of indirection, so I think it's worth it